dotnet/src/Connectors/Connectors.Memory.Redis/RedisJsonVectorStoreRecordCollection.cs

```diff
@@ -362,6 +368,53 @@ await this.RunOperationAsync(
         }
     }
 
+    /// <inheritdoc />
+    public async IAsyncEnumerable<VectorSearchResult<TRecord>> SearchAsync(VectorSearchQuery vectorQuery, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(vectorQuery);
+
+        if (this._firstVectorPropertyName is null)
```

Comment:
Should the vector property name be something that can be set in `VectorSearchQuery`? The first vectory property name could be the default if none is provided.